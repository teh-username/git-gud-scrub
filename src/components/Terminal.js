import React from 'react';

// Console design ripped from https://onlywei.github.io/explain-git-with-d3/
class Terminal extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      userInput: ''
    };

    this.onInput = this.onInput.bind(this);
    this.onSubmit = this.onSubmit.bind(this);
  }

  onInput(userInput) {
    this.setState({ userInput });
  }

  onSubmit(event) {
    event.preventDefault();
    const { userInput } = this.state;
    if (userInput) {
      this.props.handleInput(this.state.userInput);
      this.setState({ userInput: '' });
    }
  }

  render() {
    const { userInput } = this.state;
    const { entries } = this.props;
    return (
      <div className="col-lg-3 col-lg-push-3 col-sm-12 console">
        <h4>Terminal</h4>
        <div className="control-box">
          <div className="log">
            <div className="log-inner">
              <div className="info">Enter your git commands here!</div>
              <div className="entries">
                {entries.map(({ text, logLevel }, index) => (
                  <div key={index} className={logLevel}>
                    {text}
                  </div>
                ))}
              </div>
            </div>
          </div>
          <form onSubmit={this.onSubmit} className="control-box-form">
            <input
              spellCheck={false}
              type="text"
              value={userInput}
              placeholder=" enter git command"
              onChange={({ target }) => this.onInput(target.value)}
            />
          </form>
        </div>
      </div>
    );
  }
}

export default Terminal;
