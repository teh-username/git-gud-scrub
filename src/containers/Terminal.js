import { connect } from 'react-redux';

import Terminal from '../components/Terminal';
import { getEntries } from '../redux/modules/terminal';
import { executeCommand } from '../redux/modules/gitEmulator';
import { commandParser } from '../utils/commandProcessor';

const mapStateToProps = state => ({
  entries: getEntries(state)
});

const mapDispatchToProps = dispatch => ({
  handleInput: input => {
    dispatch(executeCommand(commandParser(input)));
  }
});

export default connect(mapStateToProps, mapDispatchToProps)(Terminal);
