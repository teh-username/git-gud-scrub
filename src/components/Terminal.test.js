import React from 'react';
import ReactDOM from 'react-dom';
import Terminal from './Terminal';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Terminal />, div);
  ReactDOM.unmountComponentAtNode(div);
});
