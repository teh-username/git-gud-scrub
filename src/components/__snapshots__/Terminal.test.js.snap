// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Terminal Component renders entries correctly 1`] = `
<div
  className="col-lg-3 col-lg-push-3 col-sm-12 console"
>
  <h4>
    Terminal
  </h4>
  <div
    className="control-box"
    style={
      Object {
        "height": "220px",
      }
    }
  >
    <div
      className="log"
      style={
        Object {
          "height": "195",
        }
      }
    >
      <div
        className="info"
      >
        Enter your git commands here!
      </div>
      <div
        className="entries"
      >
        <div
          className="info"
          key="0"
        >
          Darth Malak
        </div>
        <div
          className="info"
          key="1"
        >
          Darth Revan
        </div>
        <div
          className="error"
          key="2"
        >
          Darth Vader
        </div>
        <div
          className="info"
          key="3"
        >
          Darth Sidious
        </div>
      </div>
    </div>
    <form
      onSubmit={[Function]}
    >
      <input
        onChange={[Function]}
        placeholder=" enter git command"
        spellCheck={false}
        type="text"
        value=""
      />
    </form>
  </div>
</div>
`;

exports[`Terminal Component renders the console predictably 1`] = `
<div
  className="col-lg-3 col-lg-push-3 col-sm-12 console"
>
  <h4>
    Terminal
  </h4>
  <div
    className="control-box"
    style={
      Object {
        "height": "220px",
      }
    }
  >
    <div
      className="log"
      style={
        Object {
          "height": "195",
        }
      }
    >
      <div
        className="info"
      >
        Enter your git commands here!
      </div>
      <div
        className="entries"
      />
    </div>
    <form
      onSubmit={[Function]}
    >
      <input
        onChange={[Function]}
        placeholder=" enter git command"
        spellCheck={false}
        type="text"
        value=""
      />
    </form>
  </div>
</div>
`;
